memberlist.go

```diff
@@ -191,50 +191,111 @@ func (m *Memberlist) Join(existing []string) (int, error) {
 		}
 
 	}
-
 	if numSuccess > 0 {
 		errs = nil
 	}
 	return numSuccess, errs
 }
 
+// ipPort holds information about a node we want to try to join.
+type ipPort struct {
+	ip   []byte
```

Comment:
Is there any reason the type shouldn't be `net.IP`?  `[]byte` is the backing store for `net.IP` so it should be equivalent.